nextest-runner/src/test_command.rs

```diff
@@ -191,6 +195,34 @@ fn apply_package_env(cmd: &mut std::process::Command, package: &PackageMetadata<
     );
 }
 
+/// Applies environment variables spcified by the build script via `cargo::rustc-env`
+fn apply_build_script_env(cmd: &mut std::process::Command, out_dir: Utf8PathBuf) {
```

Comment:
Could you also add a unit test for this? Just testing out some of these situations.